App.tsx

```diff
@@ -1,106 +1,22 @@
-import React, { useState } from 'react';
-import { StatusBar } from 'expo-status-bar';
-import { StyleSheet, View, FlatList, Image, TouchableOpacity, TextInput } from 'react-native';
-import axios from 'axios';
-import { Ionicons } from '@expo/vector-icons';
-import ImageItem from './ImageItem';
+import React from 'react';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator, StackScreenProps } from '@react-navigation/stack';
+import PhotoGalleryScreen from './PhotoGalleryScreen';
+import PhotoDetailScreen from './PhotoDetailScreen';
+import ImageModalScreen from './ImageModalScreen';
 
-interface ImageData {
-  id: number;
-  url: string;
-}
-
-const imageData: ImageData[] = [];
-for (let i = 1; i < 70; i++) {
-  imageData.push({ id: i, url: `https://picsum.photos/id/${i}/200` });
-}
-
-export default function App() {
-  const [selectedImage, setSelectedImage] = useState<string | null>(null);
-  const [searchTerm, setSearchTerm] = useState<string>('');
-
-  const handleSearch = (text: string) => {
-    setSearchTerm(text);
-  };
-
-  const handleImageSelect = (imageUrl: string) => {
-    setSelectedImage(imageUrl);
-  };
-
-  const handleImageClose = () => {
-    setSelectedImage(null);
-  };
-
-  const renderItem = ({ item }: { item: ImageData }) => (
-    <ImageItem url={item.url} onSelect={handleImageSelect} />
-  );
-
-  const filteredImages = imageData.filter((image) => image.id.toString().includes(searchTerm));
+const Stack = createStackNavigator();
```

Comment:
Apply the type here too `const Stack = createStackNavigator<RootStackNavigator>();`